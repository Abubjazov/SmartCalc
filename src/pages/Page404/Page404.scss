@import '../../variables';

.page-404 {
	background: $main-background;
	padding-top: 200px;
	animation-name: opacity;
	animation-duration: 700ms;

	p {
		text-align: 'center';
		font-weight: 300;
		font-size: 24px;
		margin-top: 13px;
		color: $black;
	}

	a {
		display: 'block';
		text-align: center;
		font-weight: 300px;
		font-size: 24px;
		margin-top: 31px;
		color: $main-color;
	}
}
