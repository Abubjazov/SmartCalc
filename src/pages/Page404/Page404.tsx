import { Helmet } from 'react-helmet'

import { ErrorGif } from '../../components'

export const Page404 = (): JSX.Element => (
	<main className='page-404'>
		<Helmet>
			<title>404</title>
		</Helmet>
		<ErrorGif />
		<p tabIndex={0} role='alert'>
			Страница не найдена
		</p>
		<a href='/smartcalc/' role='alert'>
			Вернуться на главную
		</a>
	</main>
)
