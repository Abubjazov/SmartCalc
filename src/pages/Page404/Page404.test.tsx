import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'

import { Page404 } from './Page404'

describe('Component: Page404', () => {
	test('should render Page404', () => {
		const { asFragment } = render(
			<BrowserRouter>
				<Page404 />
			</BrowserRouter>
		)
		expect(asFragment()).toMatchSnapshot()
	})
})
