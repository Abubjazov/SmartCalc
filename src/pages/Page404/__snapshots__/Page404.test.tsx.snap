// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Page404 should render Page404 1`] = `
<DocumentFragment>
  <main
    class="page-404"
  >
    <img
      alt="An error has occurred"
      src="Error.gif"
      style="display: block; width: 128px; height: 102px; object-fit: contain; margin: 0px auto;"
    />
    <p
      role="alert"
      tabindex="0"
    >
      Страница не найдена
    </p>
    <a
      class=""
      href="/smartcalc/"
      role="alert"
    >
      Вернуться на главную
    </a>
  </main>
</DocumentFragment>
`;
